dpe/src/commands/derive_context.rs

```diff
@@ -315,9 +327,17 @@ impl CommandExecution for DeriveContextCmd {
                         &mut cert,
                     )?;
 
-                    Ok(Response::DeriveContextExportedCdi(DeriveContextExportedCdiResp {
+                    let parent_handle = if self.retains_parent() {
+                        // At this point we cannot error out anymore, so it is safe to set the updated child and parent contexts.
+                        dpe.contexts[parent_idx] = tmp_parent_context;
+                        dpe.contexts[parent_idx].handle
+                    } else {
+                        ContextHandle::new_invalid()
+                    };
```

Comment:
Should `dpe.contexts[parent_idx] = tmp_parent_context;` be unconditional? Because if `!self.retains_parent()`, we still need to mark the parent context as retuired.